### PR TITLE
Cgprod 2112 single item pause

### PR DESCRIPTION
**Pull request by FostUK**
* Remove all events created by single item mode on shutdown
* Call shutdown on overlay created
* Call single item mode on overlay closed

### CODE REVIEW

File: src/components/select/select-screen.js [hunk 1]
```diff
@@ -86,6 +86,16 @@ export class Select extends Screen {
...
 
+    addOverlay(key) {
+        this.singleItemMode.shutdown();
+        super.addOverlay(key);
+    }
+
+    _onOverlayRemoved(overlay) {
+        super._onOverlayRemoved(overlay);
+        this.singleItemMode = singleItemMode.create(this);
+    }
```
Comment (AdamXA) [inline src/components/select/select-screen.js]: Shouldn't need this stuff if we use makeAccessible - I'll have a look at a way to do this today

File: test/components/select/select-screen.test.js [hunk 2]
```diff
@@ -316,6 +342,46 @@ describe("Select Screen", () => {
         });
     });
 
+    describe("_onOverlayRemoved method", () => {
+        test("calls super method with same args", () => {
+            selectScreen._onOverlayRemoved("testOverlay");
+
+            expect(Screen.mock.instances[0]._onOverlayRemoved).toHaveBeenCalledWith("testOverlay");
+        });
```
Comment (AdamXA) [inline test/components/select/select-screen.test.js]: this test passes without the _onOverlayRemoved function in the select screen.

File: test/components/select/select-screen.test.js [hunk 3]
```diff
@@ -316,6 +342,46 @@ describe("Select Screen", () => {
...
+                bringToTop: jest.fn(),
+            };
+
+            selectScreen.singleItemMode = {
+                shutdown: jest.fn(),
+            };
+            selectScreen.addOverlay("testKey");
+
+            expect(Screen.mock.instances[0].addOverlay).toHaveBeenCalledWith("testKey");
+        });
```
Comment (AdamXA) [inline test/components/select/select-screen.test.js]: so does this one

File: src/components/select/single-item-mode.js [hunk 4]
```diff
@@ -6,46 +6,46 @@
...
 export const create = scene => {
     const continueButton = scene.layout.buttons.continue;
     if (!continueButton) return false;
 
-    a11y.removeButton(continueButton);
-    a11y.removeButton(scene.layout.buttons.next);
-    a11y.removeButton(scene.layout.buttons.previous);
-    a11y.reset();
+    const onBlur = () => scene.grid.showPage(0);
+    window.addEventListener("blur", onBlur);
```
Comment (AnthonyGoodchild) [inline src/components/select/single-item-mode.js]: This is causing undesired behaviour when switching browser tab -> switching back.
It looks like it jumps to the first cell then animates to the second without showing the asset.

Could it be related to the addition to the return in `showPage` ?
https://github.com/bbc/genie/pull/176/files#diff-b5d608f4ffc03234f3a1a5250497a26eR87

Comment (AdamXA) [reply src/components/select/single-item-mode.js]: can reproduce, will try and find out what it is now

**Pull request merged**